packages/plugin-infra/src/web3/useLookupAddress.ts

```diff
@@ -11,9 +12,18 @@ export function useLookupAddress<T extends NetworkPluginID>(
 ) {
     const chainId = useChainId(pluginId, expectedChainId)
     const { NameService, Others } = useWeb3State(pluginId)
+    const web3 = useWeb3(NetworkPluginID.PLUGIN_EVM)
 
     return useAsyncRetry(async () => {
-        if (!chainId || !domain || !Others?.isValidDomain?.(domain) || !NameService) return
-        return NameService.lookup?.(chainId, domain)
-    }, [chainId, domain, NameService, Others])
+        if (!domain || !Others?.isValidDomain?.(domain) || !web3) return
+        try {
+            if (!chainId || !NameService?.lookup) {
+                return web3.eth.ens.registry.getOwner(domain)
```

Comment:
The EVM chain-specific implementation should go here: `packages/plugins/EVM/src/state/NameService/ENS.ts` All Web3 hooks are shared across all networks.